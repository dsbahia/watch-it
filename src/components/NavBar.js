import React from "react";
import { Link } from "react-router-dom";

import "../styles/navbar.css";

function NavBar() {
  return (
    <div className="navbar">
      <div className="navbar-links-item" />

      <Link className="item-upcoming" to="/">
        Upcoming
      </Link>

      <Link className="item-trending" to="trending">
        Trending
      </Link>
    </div>
  );
}

export default NavBar;
