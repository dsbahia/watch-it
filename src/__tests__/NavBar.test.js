import React from "react";
import { MemoryRouter, Link } from "react-router-dom";
import { render } from "@testing-library/react";
import NavBar from "../components/NavBar";
import "@testing-library/jest-dom/extend-expect";

describe("NavBar", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a link with the correct className", () => {
    const { container } = render(
      <MemoryRouter>
        <Link to="/" className="item-upcoming">
          Upcoming
        </Link>
      </MemoryRouter>,
    );
    const linkElement = container.querySelector("a");

    expect(linkElement).toHaveClass("item-upcoming");
  });

  it("renders the link with the correct route", () => {
    const { container } = render(
      <MemoryRouter>
        <Link to="/" className="item-upcoming">
          {" "}
          Upcoming{" "}
        </Link>
      </MemoryRouter>,
    );
    const linkElement = container.querySelector("a");

    expect(linkElement).toHaveAttribute("href", "/");
  });
});
