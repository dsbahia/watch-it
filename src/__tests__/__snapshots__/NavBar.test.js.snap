// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <div
      class="navbar-links-item"
    />
    <a
      class="item-home"
      href="/"
    >
      Home
    </a>
    <a
      class="item-upcoming"
      href="/"
    >
      Register
    </a>
    <a
      class="item-upcoming"
      href="/"
    >
      Login
    </a>
    <div
      class="dropdown-center"
      data-bs-theme="dark"
    >
      <button
        aria-expanded="false"
        class="dropdown-toggle btn btn-secondary"
        id="movie-dropdown"
        type="button"
      >
        Movies
      </button>
    </div>
    <div
      class="dropdown-center"
      data-bs-theme="dark"
    >
      <button
        aria-expanded="false"
        class="dropdown-toggle btn btn-secondary"
        id="tv-dropdown"
        type="button"
      >
        Tv Shows
      </button>
    </div>
  </div>
</DocumentFragment>
`;
